hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/AbstractSerializationService.java

```diff
@@ -339,9 +339,14 @@ public final void writeObject(final ObjectDataOutput out, final Object obj) {
     }
 
     @Override
-    public final <T> T readObject(final ObjectDataInput in) {
+    public final <T> T readObject(final ObjectDataInput in, boolean useBigEndianForReadingTypeId) {
         try {
-            final int typeId = in.readInt();
+            final int typeId;
```

Comment:
So, now we are able to read top-level objects in this method. This is analogous to `toObject` now. I see that we have a logic to provide remote classes in that method. (See `ClassLocator.onStartDeserialization();` line)

Should we do the same here? Can we need to access remote classes in the places where we use this?

Reply:
@frant-hartm told me "snapshot really shouldn't use a class deployed by UCD" so I think we should be fine without calling that method. This was also the existing behavior. Given the javadoc I have written I suggest I don't add this.

I am reluctant the change the code here because it's on the hot path, WDYT?